packages/wallets/src/components/WalletImport.tsx

```diff
@@ -43,11 +56,58 @@ export default function WalletImport() {
     },
   });
 
-  const { fields } = useFieldArray({
+  const { fields, replace } = useFieldArray({
     control: methods.control,
     name: 'mnemonic',
   });
 
+  const submitMnemonicPaste = (mnemonicList: string) => {
+    closeMnemonicPaste();
```

Comment:
Putting `closeMnemonicPaste()` here means regardless of whether a valid mnemonic is pasted or not, dialog will be closed either.
I think dialog should remain opened if the pasted mnemonic is invalid for it is easy to re-paste again instead of re-open 'paste-mnemonic' dialog over and over.

Could you move this `closeMnemonicPaste()` after `replace(mnemonic)` ?
If you do so, it will work like this:

https://user-images.githubusercontent.com/84098616/162473238-5aa92476-007b-491d-8d81-42eb672850b3.mov